app/controllers/application_controller.rb

```diff
@@ -130,7 +130,10 @@ def skip_session_expiration
   end
 
   def set_locale
-    I18n.locale = params[:locale] || I18n.default_locale
+    I18n.locale =
+      http_accept_language.compatible_language_from(I18n.available_locales) ||
+      params[:locale] ||
+      I18n.default_locale
```

Comment:
@amoose I am working on implementing the functionality with the params in a different branch, this was just focused on the HTTP headers, but I can roll it all into this one if that is preferred? https://github.com/18F/identity-idp/compare/bh-param-i18n